### PR TITLE
[server] Name old Eigen queries to improve DataDog representation.

### PR DESCRIPTION
**Pull request by alloy**
Related to https://github.com/artsy/metaphysics/pull/1225

### CODE REVIEW

**Comment by alloy** [comment]
Dirty, but straightforward.

File: src/lib/nameOldEigenQueries.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { RequestHandler } from "express"
+
+export const nameOldEigenQueries: RequestHandler = (req, _res, next) => {
+  const agent = req.headers["user-agent"]
+  if (agent && agent.includes("Eigen")) {
+    const { query } = req.body as { query: string }
+    if (!query.startsWith("query ")) {
+      if (query.includes("saved_artworks")) {
+        // https://github.com/artsy/eigen/blob/master/Artsy/Networking/favorites.graphql
+        req.body.query = `query FavoritesQuery ${query}`
```
Comment (mzikherman) [inline src/lib/nameOldEigenQueries.ts]: oo nice!

Might we do the same thing as a catch-all in Force? Also, in Force, https://github.com/artsy/force/blob/55f6aad57d4a1b7677b358513f8c8cbb4b5d2b93/src/lib/metaphysics.coffee#L7 is where all of those made directly go thru I believe, that can also be a place to check for stragglers.

Comment (alloy) [reply src/lib/nameOldEigenQueries.ts]: What would we name those, though? It currently falls back to just the name `query`

Comment (orta) [reply src/lib/nameOldEigenQueries.ts]: hah - good hack

**Comment by alloy** [comment]
@orta Is this good to go?

**Comment by orta** [comment]
yes, merge on green

**Pull request merged**